### PR TITLE
Globbing

### PR DESCRIPTION
**Pull request by Emyrk**
Added glob opening support. It's easier when parsing templates within a directory if you can use "/*.html". 

Uses GoLang's native Globbing, so supports anything that filepath globbing supports

### CODE REVIEW

File: files/files.go [hunk 1]
```diff
@@ -123,3 +124,24 @@ func Hash(file string) (s string) {
 	}
 	return
 }
+
+// Slower than Open as it must cycle through every element in map. Open all files that match glob.
+func OpenGlob(name string) ([]io.ReadCloser, error) {
+	readers := make([]io.ReadCloser, 0)
+	for file := range staticFiles {
+		matches, err := filepath.Match(name, file)
```
Comment (bouk) [inline files/files.go]: path.Match is more appropriate here, as filepath behaves differently on windows, while the paths in staticFiles is unix-style

**Comment by Emyrk** [comment]
Changed filepath to path as Bouk recommended

File: files/files.go (hunk 1 above)
Comment (Emyrk) [inline files/files.go]: Changed in next commit as you recommended

File: files/files_dev.go [hunk 2]
```diff
@@ -42,3 +42,23 @@ func Hash(name string) (s string) {
 	io.Copy(hash, f)
 	return hex.EncodeToString(hash.Sum(nil))
 }
+
+func OpenGlob(name string) ([]io.ReadCloser, error) {
+	readers := make([]io.ReadCloser, 0)
+	for file := range staticFiles {
```
Comment (bouk) [inline files/files_dev.go]: There's no staticFiles in this file

**Comment by bouk** [comment]
Actually, thinking about it some more I don't think I'm going to accept this Pr. I think it's a bit too limited to add to the package in this state, something that provides a way to list the directories or iterate over all the files might be better suited.

If you still want to use this function without forking, you could put the openglob function in another file in the same directory as the generated file, that would work.

**Comment by Emyrk** [comment]
Ok, sounds good. Thanks for taking the time to look at it

**Pull request closed without merging**